api/upstream.go

```diff
@@ -8,14 +8,24 @@ import (
 	"github.com/google/uuid"
 )
 
-// PushedResourceIDs is a list of primary keys of resources that have been pushed to the
-// upstream
-type PushedResourceIDs struct {
-	Canaries       []string `json:"canaries,omitempty"`
-	Checks         []string `json:"checks,omitempty"`
-	Components     []string `json:"components,omitempty"`
-	ConfigItems    []string `json:"config_items,omitempty"`
-	ConfigScrapers []string `json:"config_scrapers,omitempty"`
+var TablesToReconcile = []string{
+	"components",
+	"config_scrapers",
+	"config_items",
+	"canaries",
+	"checks",
+}
+
+type PaginateRequest struct {
+	Table string    `query:"table"`
+	From  uuid.UUID `query:"from"`
+	Size  int       `query:"size"`
+}
+
+type PaginateResponse struct {
+	Hash  string    `gorm:"column:sha256sum"`
+	Next  uuid.UUID `gorm:"column:last_id"`
```

Comment:
Any reason for giving struct field and gorm column different names ?

Reply:
I was just trying to avoid aliasing to `hash` as it's a SQL keyword.